sympy/core/function.py

```diff
@@ -2185,13 +2185,25 @@ def __new__(cls, expr, variables, point, **assumptions):
         if not point:
             return sympify(expr)
 
-        # denest
-        if isinstance(expr, Subs):
-            variables = expr.variables + variables
-            point = expr.point + point
-            expr = expr.expr
-        else:
-            expr = sympify(expr)
+        def denest(expr, variables, point):
+            # Helper function to denest expressions involving Subs in some form.
+            # if expression isn't related to Subs, the expression passed is returned.
+            # Currently it can deal with expressions of type Mul(constant, Subs(...)).
+            if isinstance(expr, Subs):
+                variables = expr.variables + variables
+                point = expr.point + point
+                expr = expr.expr
+            elif isinstance(expr, Mul):
+                constant = expr.args[0]
```

Comment:
Well if we are tackling `Mul(const, Subs(..))` then I realized it would end up like this ! But yeah what you say is correct . Though even if the multiplicative part isn't a constant it'll end up similarly 
If we have the expression of the form 
```
>>> x, y, h, z = symbols('x y h z')
>>> u = Function('u')
>>> u(x + z*h, y + z*h).diff(h, 2)
```
we would end up getting `z*Subs(Derivative(u(_xi_1, _xi_2), _xi_1, _xi_2), _xi_2, h*z + y)` which is being denested fine and the answer obtained is also correct 
```

   ⎛⎛  2                 ⎞│             ⎛  2                 ⎞│
 2 ⎜⎜ ∂                  ⎟│             ⎜ ∂                  ⎟│
z ⋅⎜⎜────(u(ξ₁, h⋅z + y))⎟│           + ⎜────(u(h⋅z + x, ξ₂))⎟│           + 2⋅
   ⎜⎜   2                ⎟│             ⎜   2                ⎟│
   ⎝⎝∂ξ₁                 ⎠│ξ₁=h⋅z + x   ⎝∂ξ₂                 ⎠│ξ₂=h⋅z + y

⎛    2             ⎞│                      ⎞
⎜   ∂              ⎟│                      ⎟
⎜───────(u(ξ₁, ξ₂))⎟│                      ⎟
⎝∂ξ₂ ∂ξ₁           ⎠│ξ₂=h⋅z + y, ξ₁=h⋅z + x⎟
                                           ⎠
```
But yeah what is the best way out to handle this ?? Maybe looping through the args , taking out the subs part and keeping other things together ? Also I'm not sure how would we approach more than 2 args , like how could we deal with something like `2*z*Subs(Derivative(u(_xi_1, _xi_2), _xi_1, _xi_2), _xi_2, 2*h*z + y)` or `a***2*z*Subs(Derivative(u(_xi_1, _xi_2), _xi_1, _xi_2), _xi_2, 2*h*z + y)` and other random cases . I'll give this a try and check if I find something substantial !